### DISCUSSION TITLE
LibreOffice

### DISCUSSION THREAD

**Discussion by raffaelemancuso** [hyprwm/Hyprland#1027] [Q&A] [answered]
I don't understand why, but when I open LibreOffice, hyprland enters a new "mode": the LibreOffice window takes all the screen, and I can still switch between open apps in the workspace with the movefocus dispatcher, and each app will still take the entire window.

How can I have LibreOffice behave like any other app? (so be tiled).

Also, how can I have hyprland switch to this new "mode" in general?

I don't have any LibreOffice thing in my config file. So it's not on me.

**Comment by ElJeffe** [comment]
It seems to behave correctly here:
![1668588323-wayshot](https://user-images.githubusercontent.com/54236/202132540-fbde442b-7467-4eb1-97c1-9b4815399f2e.png)

FWIW there are two fullscreen modes in Hyprland:

```
hyprctl dispatch fullscreen 0
hyprctl dispatch fullscreen 1
```

Do you mean one of these with the new mode?

**Reply by raffaelemancuso** [reply]
Yes, basically I entered in `hyprctl dispatch fullscreen 1` whenever `soffice` was called.

Now I have run `hyprctl dispatch fullscreen 0` with `soffice` open, and the Libreoffice window is tiled again. Furthermore, it does *not* enter `hyprctl dispatch fullscreen 1` automatically when `soffice` is called anymore. I don't know why????

There seems to be a third mode, which is the "fullscreen" dispatcher, like in:
```
bind = $mainMod, U, fullscreen,
```
The difference is that in this mode, also the space of the bar is taken.

**Reply by ElJeffe** [reply]
> There seems to be a third mode, which is the "fullscreen" dispatcher, like in:
> 
> ```
> bind = $mainMod, U, fullscreen,
> ```
> 
> The difference is that in this mode, also the space of the bar is taken.

This is the mode you get when `hyprctl dispatch fullscreen 0` is called on a non-fullscreen window. If the window is already fullscreen, it will exit fullscreen mode. The `0` here does not mean 'no fullscreen' but 'fullscreen without bar' as described https://wiki.hyprland.org/Configuring/Dispatchers/

**Reply by raffaelemancuso** [reply]
Thank you very much. I am now understanding how that dispatcher works.

I find both its name and its doc a bit confusing confusing.

I think it should be named "togglefullscreen", to be consistent with "togglefloating", "togglegroup", etc.

And I think the doc should read something like:
```
0 - toggle between tiled and fullscreened (takes your entire screen), 
1 - toggle between tiled and maximized (keeps gaps and bar(s))
```

**Comment by vaxerski** [comment, marked as answer]
LO for some reason requests a maximize on open, for some people including me. That's what's happening.

**Reply by raffaelemancuso** [reply]
Is there something I can put in my config file to prevent that?

**Reply by ElJeffe** [reply]
you can try

`windowrulev2 = tile,class:^(libreoffice.*)$`

to force tile libreoffice windows on start.

**Reply by kepi** [reply]
This doesn't work, at least not in current version. Libreoffice take fullscreen anyway. Using `nomaximizerequest` mentioned bellow helped.

**Comment by theotrst** [comment]
I had the same problem. I solved it by adding

`windowrulev2 = nofullscreenrequest,class:^(libreoffice.*)$`

**Reply by MightyPlaza** [reply]
this will not work anymore
use the new rule "nomaximizerequest"
same issue was happening on mpv and is happening in firefox
I recommend you open an issue on their bug tracker if you want this issue actually solved
